src/components/createTodo/index.tsx

```diff
@@ -0,0 +1,87 @@
+// import { zodResolver } from '@hookform/resolvers/zod';
+// import { useForm } from 'react-hook-form';
+import { useRef, useState } from 'react';
+
+import AddIcon from '@/assets/icons/AddIcon';
+import Dialog from '@/components/dialog';
+import useKabanaStore from '@/stores/store';
+
+import Tag from '../tag';
+import colorList from './colorList';
+const CreateTodo = () => {
+  const modalIsOpen = useKabanaStore((state) => state.createTodo);
+  const toggleModal = useKabanaStore((state) => state.toggleCreateTodo);
+  const [tagList, setTagList] = useState<string[]>([]);
+  const [addedColors, setAddedColors] = useState<string[]>([]);
+  const tagRef = useRef<HTMLInputElement>(null);
+
+  const createRandomNumber = () => {
+    const result = Math.floor(Math.random() * colorList.length);
+    return result;
+  };
+
+  const createDeleteTags = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      const randomNum = createRandomNumber();
+      if (tagList.includes(e.currentTarget.value)) {
+        e.currentTarget.value = '';
+      }
+
+      if (addedColors.indexOf(colorList[randomNum]) === -1) {
+        setAddedColors([...addedColors, colorList[randomNum]]);
+      }
+      setTagList([...tagList, e.currentTarget.value]);
+      e.currentTarget.value = '';
+      e.currentTarget.focus();
+    }
+    if (e.key === 'Backspace' && e.currentTarget.value !== null) {
```

Comment:
`e.currentTarget.value === ''` 이렇게 안하고 아래처럼 null로 해도 정상 동작하나요?
```suggestion
    if (e.key === 'Backspace' && e.currentTarget.value !== null) {
    ...
    }
```

Reply:
요기는 null로 지정하니까 Type 'null' is not assignable to type 'string'. 이렇게 뜨더라구여... 그래서 일단 빈문자열로 해놨습니당